File: Project_3_Elite-Bank/Elite-Bank-Backend/EliteBank_UserBankMicroservice/src/main/java/com/userbankdetails/EliteBank_UserBankMicroservice/User/Model/UserEntity.java
package com.userbankdetails.EliteBank_UserBankMicroservice.User.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.userbankdetails.EliteBank_UserBankMicroservice.Account.Model.AccountEntity;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.Set;

@Entity
@Table(name = "UserDetails")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
public class UserEntity implements UserDetails {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name="id")
    private String user_id;
    private String account_holder_first_name;
    private String account_holder_last_name;
    private String user_password;
    private String role;
    private String active;
    private String account_holder_address;
    private String account_holder_city;
    private String account_holder_state;
    private String account_holder_country;
    private String account_holder_pincode;
    private String account_holder_phone_no;

    @Column(unique = true)
    private String account_holder_email;
    private String account_holder_dob;
    private String account_holder_gender;
    private String account_holder_aadhar_no;
    private String account_holder_pan_no;
    private String account_holder_photo;
    private String account_holder_marital_status;
    private String account_holder_religion;
    private String account_holder_category;
    private String account_holder_occupation;
    private String account_holder_qualification;
    private String account_holder_staff_of_bank;
    private String account_holder_residency_status;
    private String account_holder_gross_income;

    @ManyToOne(fetch = FetchType.EAGER,cascade = CascadeType.ALL)
    private AccountEntity accountdata;


    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return null;
    }

    @Override
    public String getPassword() {
        return this.user_password;
    }

    @Override
    public String getUsername() {
        return this.account_holder_email;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: Project_3_Elite-Bank/Elite-Bank-Frontend/src/app/Core/Model/UserEntity.ts
export interface UserEntity {
  user_id?: String;
  active?: String;
  role?: String;
  account_holder_name?: String;
  account_holder_address?: String;
  account_holder_city?: String;
  account_holder_state?: String;
  account_holder_country?: String;
  account_holder_pincode?: String;
  account_holder_phone_no?: String;
  account_holder_email?: String;
  account_holder_dob?: String;
  account_holder_gender?: String;
  account_holder_aadhar_no?: String;
  account_holder_pan_no?: String;
  account_holder_photo?: String;
  account_holder_marital_status?: String;
  account_holder_religion?: String;
  account_holder_category?: String;
  account_holder_occupation?: String;
  account_holder_qualification?: String;
  account_holder_staff_of_bank?: String;
  account_holder_residency_status?: String;
  account_holder_gross_income?: String;
  accountdata?: {
    account_table_id?: String;
    customer_id?: String;
    account_number?: String;
    account_ifsccode?: String;
    account_type?: String;
    account_balance?: String;
    account_total_loan?: String;
    account_opening_date?: String;
  };
  statementEntityList?: [];
  debitCardEntityList?: [];
  creditCardEntityList?: [];
  chart_data?: {
    yearlyData?: [];
    expenseData?: [];
  };
}
